src/utils.js

```diff
@@ -873,43 +873,34 @@ export function toIdSchema(
 
 export function toPathSchema(schema, name = "", definitions, formData = {}) {
```

Comment:
Are you changing the functionality of toPathSchema in some way or just refactoring it?

Reply:
I am changing the functionality of `toPathSchema`. Previously, if you had

```
schema = {
    type: "object",
    properties: {
        values: {
            type: "array",
            items: {
                type: "string"
            }
        }
    }
}

formData = {
    values: ["asdf", "ghkj"]
}
```

`toPathSchema(schema, formData = formData)` would not give the array a `$name` key, and would be

```
pathSchema = {
    $name: "",
    values: {
        0: {
            $name: "values.0"
        },
        1: {
            $name: "values.1"
        }
    }
}
```

My change results in

```
pathSchema = {
    $name: "",
    values: {
        $name: "values",
        0: {
            $name: "values.0"
        },
        1: {
            $name: "values.1"
        }
    }
}
```

which I believe is more consistent since there's a `$name` for ALL nesting. Again, I made this functionality change because it looks like `toPathSchema` is only used for omitting extra data and does not break anything else. This makes it easier to traverse the path schema and get the path if the array is empty.

Reply:
Got it, thanks!